### PR TITLE
Remove old re-partitioning functions

### PR DESCRIPTION
**Pull request by marcocitus**
DESCRIPTION: Removes deprecated re-partitioning functions like worker_hash_partition_table

This PR:
- Removes re-partitioning functions that were only kept in 11.0 for rolling upgrades
- Removes the check-worker schedule which was mainly testing these functions
- Moves commonly used utility function from the files containing re-partitioning logic to dedicated files in utils/
- Bumps the master branch version to 11.1

### CODE REVIEW

File: src/backend/distributed/utils/array_type.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#include "utils/lsyscache.h"
+
+
+/*
+ * DeconstructArrayObject takes in a single dimensional array, and deserializes
+ * this array's members into an array of datum objects. The function then
+ * returns this datum array.
+ */
+Datum *
+DeconstructArrayObject(ArrayType *arrayObject)
```
Comment (marcocitus) [inline src/backend/distributed/utils/array_type.c:26]: there's no new code in this PR, these were moved from worker_*.c files

File: src/backend/distributed/shared_library_init.c [hunk 2]
```diff
@@ -437,7 +438,7 @@ _PG_init(void)
 static void
 DoInitialCleanup(void)
 {
-	RepartitionCleanupJobDirectories();
+	CleanupJobCacheDirectory();
```
Comment (onderkalaci) [inline src/backend/distributed/shared_library_init.c:441]: minor: the function comment is outdated?

File: src/include/distributed/utils/array_type.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#ifndef CITUS_ARRAY_TYPE_H
+#define CITUS_ARRAY_TYPE_H
+
+#include "postgres.h"
+
+#include "utils/array.h"
+
+
+extern Datum * DeconstructArrayObject(ArrayType *arrayObject);
+extern int32 ArrayObjectCount(ArrayType *arrayObject);
```
Comment (onderkalaci) [inline src/include/distributed/utils/array_type.h:20]: minor: maybe move `DatumArrayToArrayType` function to here as well 
```
src/include//distributed/listutils.h:extern ArrayType * DatumArrayToArrayType(Datum *datumArray, int datumCount,

```

**Comment by onderkalaci** [comment]
for future reference, this can be removed because the old logic is removed via https://github.com/citusdata/citus/pull/5400

File: src/backend/distributed/worker/worker_partition_protocol.c [hunk 4]
```diff
@@ -2,1304 +2,42 @@
...
-							partitionColumnType,
-							&RangePartitionId, (const void *) partitionContext,
-							partitionFileArray, fileCount);
-
-	/* close partition files and atomically rename (commit) them */
-	ClosePartitionFiles(partitionFileArray, fileCount);
-	CitusRemoveDirectory(taskDirectory->data);
-	RenameDirectory(taskAttemptDirectory, taskDirectory);
-	PG_RETURN_VOID();
+	ereport(ERROR, (errmsg("this function is deprecated and only kept for testing "
```
Comment (onderkalaci) [inline src/backend/distributed/worker/worker_partition_protocol.c:29]: Minor: if that's simple enough, maybe we move the 8 UDFs you remove to single file like `worker_partition_protocol` or such? 

Given that we won't be dropping these soon, could be nice to document all there. Obviously optional suggestion

File: src/backend/distributed/utils/directory.c [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+						errmsg("could not create directory \"%s\": %m",
+							   directoryName->data)));
+	}
+}
+
+
+/*
+ * FileIsLink checks whether a file is a symbolic link.
+ */
+#ifdef WIN32
```
Comment (onderkalaci) [inline src/backend/distributed/utils/directory.c:73]: I guess we cannot configure on windows, but maybe we remove WIN32 defs separately

**Comment by onderkalaci** [comment]
> DESCRIPTION: Removes deprecated re-partitioning functions like worker_hash_partition_table

Should we also add one DESCRIPTION for the dropped GUCs?

File: src/backend/distributed/commands/utility_hook.c [hunk 6]
```diff
@@ -427,42 +426,6 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 		}
 	}
 
-	/*
-	 * TRANSMIT used to be separate command, but to avoid patching the grammar
-	 * it's now overlaid onto COPY, but with FORMAT = 'transmit' instead of the
-	 * normal FORMAT options.
-	 */
-	if (IsTransmitStmt(parsetree))
```
Comment (onderkalaci) [inline src/backend/distributed/commands/utility_hook.c:435]: Ah, I had noted this but you realized before I pushed my comment :) Great to see we remove

File: src/backend/distributed/executor/transmit.c [hunk 7]
```diff
@@ -310,113 +312,3 @@ ReceiveCopyData(StringInfo copyData)
...
-	return false;
-}
-
-
-/*
- * TransmitStatementUser extracts the user attribute from a
- * COPY ... (format 'transmit', user '...') statement.
- */
-char *
-TransmitStatementUser(CopyStmt *copyStatement)
```
Comment (onderkalaci) [inline src/backend/distributed/executor/transmit.c:344]: I guess we still use the word `transmit`? If we decide to change (to `copy_utils.c` or such), maybe we rename `FileOpenForTransmit` function in this file as well?

File: src/backend/distributed/utils/directory.c [hunk 8]
```diff
@@ -0,0 +1,215 @@
+/*-------------------------------------------------------------------------
+ *
+ * directory.c
```
Comment (onderkalaci) [inline src/backend/distributed/utils/directory.c:3]: I guess all the functions here copy & paste as well?

**Review by onderkalaci — APPROVED** [review]
I don't really have any major comments, we are only removing code. I checked for the removed tests just in case we'd want to keep some, but all seems fine to remove.

thanks for removing these! As soon as enterprise conflict is resolved, we should be good to merge

File: src/backend/distributed/utils/directory.c (hunk 8 above)
Comment (marcocitus) [reply src/backend/distributed/utils/directory.c:3]: yes

File: src/backend/distributed/utils/directory.c (hunk 5 above)
Comment (marcocitus) [reply src/backend/distributed/utils/directory.c]: removed

File: src/include/distributed/utils/array_type.h (hunk 3 above)
Comment (marcocitus) [reply src/include/distributed/utils/array_type.h:20]: moved

File: src/backend/distributed/shared_library_init.c (hunk 2 above)
Comment (marcocitus) [reply src/backend/distributed/shared_library_init.c:441]: updated

**Pull request merged**